pyproject.toml

```diff
@@ -6,6 +6,7 @@ requires = [
   "setuptools == 68.1.0",
   "setuptools_scm[toml] == 7.1.0",
   "frequenz-repo-config[api] == 0.10.0",
+  "grpcio == 1.66.2",
```

Comment:
You should probably also pin `grpcio-tools` and `protobuf`, if it is working now, I would do it at least just in case for the future, as by how protobuf works, they should be pinned too.

I would also add the comments we have in other repos, otherwise it could be very easy for someone that doesn't know about the intricancies of protobuf/grpcio to feel tempted to remove the pins, widen them or whatever, it is not obvious at all what's going on with that, and how it is also related to the min runtime dependencies.

Example (https://github.com/frequenz-floss/frequenz-api-common/blob/8c3fc4fe42357b776f363722cda374116484d174/pyproject.toml#L9-L15)
```
  # We need to pin the protobuf, grpcio and  grpcio-tools dependencies to make
  # sure the code is generated using the minimum supported versions, as older
  # versions can't work with code that was generated with newer versions.
  # https://protobuf.dev/support/cross-version-runtime-guarantee/#backwards
  "protobuf == 5.28.0",
  "grpcio-tools == 1.66.1",
  "grpcio == 1.66.1",
```

and (https://github.com/frequenz-floss/frequenz-api-common/blob/8c3fc4fe42357b776f363722cda374116484d174/pyproject.toml#L35-L44):

```
dependencies = [
  # We can't widen beyond the current value unless we bump the minimum
  # requirements too because of protobuf cross-version runtime guarantees:
  # https://protobuf.dev/support/cross-version-runtime-guarantee/#major
  "protobuf >= 5.28.0, < 7", # Do not widen beyond 7!
  # We couldn't find any document with a spec about the cross-version runtime
  # guarantee for grpcio, so unless we find one in the future, we'll assume
  # major version jumps are not compatible
  "grpcio >= 1.66.1, < 2", # Do not widen beyond 2!
```

Reply:
https://github.com/frequenz-floss/frequenz-api-electricity-trading/pull/107